### ISSUE TITLE
cannot unwrap db

### ISSUE THREAD

**Issue by retorquere** [jakearchibald/idb#323]
given this code

```
import { unwrap, openDB, DBSchema, IDBPDatabase } from 'idb'

interface Schema extends DBSchema {
  test: {
    value: number
    key: number
  }
}

async function main() {
  const db: IDBPDatabase<Schema> = await openDB('test', 1, {
    upgrade($db, oldVersion, newVersion, transaction, event) {
      $db.createObjectStore('test')
    },
  })

  unwrap(db).close()
}
```

during typescript compilation, I am getting

```
content/db/testidb.ts(17,10): error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type 'IDBPDatabase<Schema>' is not assignable to parameter of type 'Promise<unknown>'.
      Type 'IDBPDatabase<Schema>' is missing the following properties from type 'Promise<unknown>': then, catch, finally, [Symbol.toStringTag]
content/db/testidb.ts(17,14): error TS2339: Property 'close' does not exist on type 'IDBCursorWithValue'.
```

**Comment by Oleksii20in** [comment]
?😅

**Comment by jakearchibald** [comment]
Fixed in bfe838e044aa8f9920956d77dd96365893fdca61, released as 8.0.1
